### ISSUE TITLE
Code doesn't report SA1304

### ISSUE THREAD

**Issue by GregReddick** [DotNetAnalyzers/StyleCopAnalyzers#1780]
First reported in #1766, Issue D11, with refinements here.

The code below would report SA1304, SA1307, and SA1311 in classic StyleCop. In the current analyzer, it reports only SA1307 and SA1311, not reporting SA1304. I think even two warnings may be too much, and we should at the least document the difference, and consider suppressing SA1307 too. If there is a large code base trying to be changed to the StyleCop style, the proliferation of warnings becomes daunting when you have 1000+ warnings that show up, when half of which are redundant reporting of "capitalize your field names".

```
namespace RuleViolations
{
    /// <summary>A rule SA1311 fail.</summary>
    public static class RuleSA1311Fail
    {
        /// <summary>The value.</summary>
        public static readonly int value = 1;
    }
}
```

**Comment by sharwell** [comment]
Related to #574 and #1306.

I think only SA1311 should be reported here.

**Comment by sharwell** [comment]
I remember why we did this now.

* SA1307 affects accessible fields, which means the public API surface area.
* SA1311 is a style choice which partially overlaps with SA1307.

It is reasonable to think some teams would disable SA1307 while others would disable SA1311. If SA1311 was a strict subset of SA1307, then it would make sense to not report SA1307 here, but that is not the case.
